src/main/java/com/softserve/demo/service/impl/ServiceFromProvidersImpl.java

```diff
@@ -1,35 +1,33 @@
 package com.softserve.demo.service.impl;
 
+import com.softserve.demo.model.Offer;
 import com.softserve.demo.model.Service;
 import com.softserve.demo.repository.ServicesRepository;
 import com.softserve.demo.service.ServiceFromProviders;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.data.domain.Page;
-import org.springframework.data.domain.PageRequest;
 
 import java.util.List;
 
 @org.springframework.stereotype.Service
 public class ServiceFromProvidersImpl implements ServiceFromProviders {
 
-    @Autowired
-    ServicesRepository servicesRepository;
+    private ServicesRepository servicesRepository;
 
+    public ServiceFromProvidersImpl(ServicesRepository servicesRepository) {
+        this.servicesRepository = servicesRepository;
+    }
 
     @Override
-    public void createService(Service service) {
-        servicesRepository.save(service);
+    public Service createService(Service service) {
+        return servicesRepository.save(service);
     }
 
     @Override
-    public Service updateService(Integer id, Service service) {
-        if (servicesRepository.existsById(id)) {
-            Service serviceFromDB = servicesRepository.getOne(id);
-            serviceFromDB.setServiceName(service.getServiceName());
-            servicesRepository.save(serviceFromDB);
-            return servicesRepository.getOne(id);
+    public Service updateService(Service service) {
+        Service serviceFromDB = servicesRepository.getOne(service.getId());
+        if (serviceFromDB == null) {
```

Comment:
Maybe exists by id here would be better. Because we don`t use serviceFromDB object.

Reply:
I fixed it.